### ISSUE TITLE
recap-query endpoint, if given null item, finds the world

### ISSUE THREAD

**Issue by johnhawkinson** [freelawproject/courtlistener#859]
Or some random fraction of it.

This query is reasonable:

```
curl -v -H 'Authorization: Token REDACTED' 'https://www.courtlistener.com/api/rest/v3/recap-query/?docket_entry__docket__court=mad&pacer_doc_id__in=09505567375'
```

and returns no documents.

```
{"count":0,"next":null,"previous":null,"results":[]}
```

 But if a stray comma is added:

```
curl -v -H 'Authorization: Token REDACTED' 'https://www.courtlistener.com/api/rest/v3/recap-query/?docket_entry__docket__court=mad&pacer_doc_id__in=09505567375,'
```

Then it returns a whole slew of null documents:

```
{"count":206,"next":null,"previous":null,"results":[{"pacer_doc_id":"","filepath_local":"recap/gov.uscourts.mad.100734.1106.1.pdf","id":14805337},{"pacer_doc_id":"","filepath_local":"recap/gov.uscourts.mad.100734.1115.1.pdf","id":14805349},{"pacer_doc_id":"","filepath_local":"recap/gov.uscourts.mad.108516.1.1.pdf","id":22030133},{"pacer_doc_id":"","filepath_local":"recap/gov.uscourts.mad.108516.9.2.pdf","id":22030146},{"pacer_doc_id":"","filepath_local":"recap/gov.uscourts.mad.108516.9.1.pdf","id":22030143},{"pacer_doc_id":"","filepath_local":"recap/gov.uscourts.mad.108516.16.1.pdf","id":22030160},
...
```

and when I queried with several hundred documents (of which only 3 were available) and inadvertently added a comma, instead I got 5263 results, inexplicably:

```
{"count":5263,"next":"https://www.courtlistener.com/api/rest/v3/recap-query/?pacer_doc_id__in=09505567375%2C09505567394%2C09505567375%2C09505567395%2C09505567396%2C...&page=2","previous":null,"results":[{"pacer_doc_id":"","filepath_local":"recap/gov.uscourts.nyed.421076/gov.uscourts.nyed.421076.1.0.pdf","id":37296924},{"pacer_doc_id":"","filepath_local":"recap/gov.uscourts.nyed.409340/gov.uscourts.nyed.409340.55.0.pdf","id":34584126},{"pacer_doc_id":"","filepath_local":"recap/gov.uscourts.nyed.389141/gov.uscourts.nyed.389141.62.0.pdf","id":31918329},{"pacer_doc_id":"","filepath_local":"recap/gov.uscourts.nyed.375147/gov.uscourts.nyed.375147.229.0.pdf","id":37192882},{"pacer_doc_id":"","filepath_local":"recap/gov.uscourts.nyed.421077/gov.uscourts.nyed.421077.1.0.pdf","id":37322763},{"pacer_doc_id":"","filepath_local":"recap/gov.uscourts.nyed.382698/gov.uscourts.nyed.382698.43.0.pdf","id":37250282}
```

Oh I see, right, because there's no Court in there for some reason.

Anyhow, I think this behavior is undesirable and should probably not be permitted, although it's unlikely that many will run into it.

Hrmm, Just ran it and got 5260 results.

Also...what are these items with no `pacer_doc_id` and how did they get there?

**Comment by mlissner** [comment]
This is certainly something deep in the bowels of Django REST Framework, and it seems like it makes sense to me. It's splitting on the comma, then filtering for anything with doc ID or NULL. This might not be your intent, but it's a logical thing to do at least. But if we really think this is a bug, I guess we should take it to Django REST Framework to see if they'll accept a fix. 

> Also...what are these items with no pacer_doc_id and how did they get there?

Honestly, I forget, but we've got a lot of them. We even have a way of generating URLs for them on PACER that uses their document number and case ID, I think. But I forget how they happen, alas. Might just be something from the OG RECAP. 

Anyway, I'm inclined to close this one.

**Comment by johnhawkinson** [comment]
Oh, I agree that the API is returning what is queried for. I just think that is not a query that should be allowed, similar to why one cannot ask Google to return all documents. However, maybe this is not a real issue because of pagination (although response time was...not fast, which makes me suggest a conversation with your DBA may be in order).

So, I think:
* Ensure that there is no database performance impact to permitting this query.
* Repair any database indexing failure or other DBA issue.
* If appropriately performant, continue to allow this relatively useless query.

**Comment by mlissner** [comment]
Yeah, I'm not sure what's up with these. We've got all the right fields properly indexed, but performance isn't great. This is similar to the issue that was causing downtime earlier this week. Maybe the DB just needs more memory, as in #385.

**Comment by johnhawkinson** [comment]
If there's one thing I'm confident of, it is that the database does not "just need more memory."
That's throwing resources at the problem.
It may be the easiest approach, but it's not the Right one. Something else is actually wrong.

**Comment by mlissner** [comment]
Go on?

**Comment by johnhawkinson** [comment]
Well, I don't know what it is! Hence "one thing I'm confident of," not two!
